### PR TITLE
Paillier

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: paillier/paillier.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+		// random two primes p and q
+		p, err := rand.Prime(rand.Reader, pqSize)
+		if err != nil {
+			return nil, nil, err
+		}
+		q, err := rand.Prime(rand.Reader, pqSize)
+		if err != nil {
+			return nil, nil, err
+		}
+
```
Comment (cychuang0924) [inline paillier/paillier.go]: Due to correct, maybe need to check gcd(p,q)=1. However, in this case, we consider that p and q are primes with equal lengths, so only need to check "p !=q". 
In fact, the probability of p=q happened is very low about 2^(-pqSize). 
In our application, this value is 2^(-1024).



Well, this is my mistake.

Comment (markya0616) [reply paillier/paillier.go]: OK, so we just meed to check if p == q.

Comment (cychuang0924) [reply paillier/paillier.go]: Yes !

**Comment by cychuang0924** [comment]
Nice work!

**Pull request merged**